components/org.wso2.carbon.identity.idp.metadata.saml2/src/main/java/org/wso2/carbon/identity/idp/metadata/saml2/util/SAMLMetadataConverter.java

```diff
@@ -78,22 +80,41 @@ public boolean canHandle(Property property) {
         }
     }
 
+    /**
+     * Retrieves whether the SAML IDP is available in registry or not.
+     *
+     * @param tenantId ID of the tenant domain
+     * @param idPName  ID of the IDP which is needed to be checked whether it has a metadata string in registry
+     * @return a boolean regarding to the condition
+     * @throws IdentityProviderManagementException if there is an error in the registry access
+     */
+    public boolean canDelete(int tenantId, String idPName) throws IdentityProviderManagementException {
+
+        try {
+            UserRegistry registry = IDPMetadataSAMLServiceComponentHolder.getInstance().getRegistryService()
+                    .getGovernanceSystemRegistry(tenantId);
+            return registry.resourceExists(IdentityRegistryResources.SAMLIDP + idPName);
+        } catch (RegistryException e) {
+            throw new IdentityProviderManagementException("Error while checking the registry for the IDP.", e);
+        }
+    }
 
     /**
-     * Returns a FederatuedAuthenticatorConfigObject that is generated using metadata
+     * Returns a FederatedAuthenticatorConfigObject that is generated using metadata
      *
-     * @param properties,builder
+     * @param properties, builder
      * @return FederatedAuthenticatorConfig
      * @throws javax.xml.stream.XMLStreamException, IdentityProviderManagementException
      */
-    public FederatedAuthenticatorConfig getFederatedAuthenticatorConfig(Property properties[], StringBuilder builder) throws javax.xml.stream.XMLStreamException, IdentityProviderManagementException {
-
+    public FederatedAuthenticatorConfig getFederatedAuthenticatorConfig(Property[] properties, StringBuilder builder)
+            throws javax.xml.stream.XMLStreamException, IdentityProviderManagementException {
 
         String metadata = "";
-        for (int y = 0; y < properties.length; y++) {
+        for (Property property : properties) {
 
-            if (properties[y] != null && properties[y].getName() != null && properties[y].getName().toString().equals("meta_data_saml")) {
-                metadata = properties[y].getValue();
+            if (property != null && property.getName() != null &&
```

Comment:
Does `property.getName()` return a String? If so we can invert equals method so that we can avoid the null check.